Use generics on Reader

lcobucci opened the pull request:
Allowing for static analysis tools to more easily understand things.

lcobucci committed:
Use generics on Reader

Allowing for static analysis tools to more easily understand things.

lcobucci commented on lib/Doctrine/Common/Annotations/Reader.php:
```diff
@@ -26,9 +26,11 @@ public function getClassAnnotations(ReflectionClass $class);
-     * @return object|null The Annotation or NULL, if the requested annotation does not exist.
+     * @return T|null The Annotation or NULL, if the requested annotation does not exist.
+     *
+     * @template T
```
This is a very unusual place for `@template` :smile:

lcobucci replied:
Perhaps https://github.com/lcobucci/coding-standard/blob/7.0.x/src/Lcobucci/ruleset.xml#L17 can help?

beberlei commented:
@lcobucci you can target this for 1.11 imho, its only changing comments. We did the same with Psalm improvements on ORM 2.7

lcobucci commented on lib/Doctrine/Common/Annotations/TokenParser.php:
```diff
@@ -12,6 +12,8 @@
 use const T_AS;
 use const T_COMMENT;
 use const T_DOC_COMMENT;
+use const T_NAME_FULLY_QUALIFIED;
```
These issues were in the repo... but I guess CI didn't catch them before? Weeeird

lcobucci replied:
Not on 1.11.x, though :laughing:

lcobucci commented:
> @lcobucci you can target this for 1.11 imho, its only changing comments. We did the same with Psalm improvements on ORM 2.7

@beberlei done!

beberlei commented:
Yeah #372 has the same phpstan error, no clue how to work aorund that, i cant reproduce locally.

lcobucci committed:
Ignore errors due to missing properties

This test is making sure we throw exceptions on missing properties.
PHPStan doesn't need to alert us here.

lcobucci commented:
> Yeah #372 has the same phpstan error, no clue how to work aorund that, i cant reproduce locally.

@beberlei I was able to reproduce it locally... squeezed an extra commit in, explaining why it's okay to ignore these errors :+1:

beberlei commented:
@lcobucci awesome, found the same workaround :D https://github.com/doctrine/annotations/pull/372/commits/2f69b8df153b4b60f9e580d002a53680c4d4576e
